builder.md

```diff
@@ -1,57 +1,180 @@
-# Builder Guidelines
-**Version 0.10.0**
+# Base16 builder specification
+**Version 0.11.0**
 
-A base16 builder is an application that can build syntax highlighting definition files for text editors by using base16 scheme files which contain a collection of colours and base16 template files which contain syntax highlighting rules.
+*The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).*
 
-Builders are designed for theme maintainers' ease of use. Theme maintainers should provide built versions of their theme so the end user doesn't need to be aware of the builder.
+## 1. Introduction
 
-## File Structure
+"Building" refers to replacing placeholders on base16 _[template
+files](./file.md)_ with colors from base16 _[schemes](./styling.md)_.
 
-### Schemes Repository
+A template is a mustache file that acts as a "blueprint": it represents how to
+output the scheme into other file formats, or (more commonly) how to output a
+configuration file to theme a specific software. For example: a [vim
+template](https://github.com/base16-project/base16-vim/blob/main/templates/default.mustache).
 
-The schemes repo should either be stored in a common location (perhaps referred to by environment variable or command line flag) or dynamically embedded in the builder.
+A scheme is a yaml file that represents a palette of 16 colors: to be replaced
+on the placeholders the template files hold. For example: the [solarized
+scheme](https://github.com/base16-project/base16-schemes/blob/main/solarized-dark.yaml)
 
-- `/*.yaml`
+A base16 builder is, essentially, an application that somehow implements the
+_building_ feature specification: filling the placeholders on templates with
+scheme colors.
 
-### Template Repository
+Builders are categorized into:
 
-Each template repository should have a templates folder containing a config.yaml and any needed mustache template files.
+- _reference tooling_: These follow the spec closely and are
+  designed for unopinionated lower-level ("plumbing") usage, specifically for
+  scripting and as component to build more complex base16 applications. High
+  quality reference builders will usually be adopted by the base16-project
+  organization;
+- _extended tooling_: These are built for end users and are usually tailored
+  for a usecase. This includes GUI apps, ergonomical CLI tools, scheme
+  managers, web apps, and more. These apps are encouraged to use (or create)
+  reference implementations as libraries, but may also implement the builder
+  feature themselves. All of the following `MUST`s can be replaced by `SHOULD`s
```

Comment:
Nit: I would argue that only CLI-related MUSTs can be replaced. If extended tooling implements the template variable spec, all those need to still be MUST.